File: treebolicClientsIface/src/main/java/org/treebolic/clients/iface/ITreebolicClient.java
/*
 * Copyright (c) Treebolic 2019. Bernard Bou <1313ou@gmail.com>
 */

package org.treebolic.clients.iface;

import android.content.Intent;

/**
 * Interface to client
 *
 * @author Bernard Bou
 */
public interface ITreebolicClient
{
	/**
	 * Connect
	 */
	void connect();

	/**
	 * Disconnect
	 */
	void disconnect();

	/**
	 * Request model from source
	 *
	 * @param source    source
	 * @param base      document base
	 * @param imageBase base image base
	 * @param settings  settings
	 * @param forward   forward intent
	 */
	void requestModel(String source, String base, String imageBase, String settings, Intent forward);
}

File: treebolicClientsIface/src/main/java/org/treebolic/clients/iface/ITreebolicClient.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.treebolic.clients.iface

import android.content.Intent

/**
 * Interface to client
 *
 * @author Bernard Bou
 */
interface ITreebolicClient {

    /**
     * Connect
     */
    fun connect()

    /**
     * Disconnect
     */
    fun disconnect()

    /**
     * Request model from source
     *
     * @param source    source
     * @param base      document base
     * @param imageBase base image base
     * @param settings  settings
     * @param forward   forward intent
     */
    fun requestModel(source: String, base: String?, imageBase: String?, settings: String?, forward: Intent?)
}
